fix: (platform) Link documentation changes and Extending from Base class.

DeepakSap14 opened the pull request:
#### Please provide a link to the associated issue.
mentioned in https://github.com/SAP/fundamental-ngx/issues/2542

#### Please provide a brief summary of this pull request.
1). It has documentation details updated.
2). added type attribute for setting native type attribute on <a> tag.
3). Added examples for added attributes.

BREAKING CHANGES:
Renamed properties:
Before :type After: linktype

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] Documentation Examples
- [x] Stackblitz works for all examples

kavya-b commented on apps/docs/src/app/platform/component-docs/platform-link/platform-link-header/platform-link-header.component.html:
```diff
@@ -1,8 +1,12 @@
-    <br /><br />
-    Link can be clicked by keyboard enter key and Mouse click.
+    Link in Platform is functionally similar to the Link in
+    <a href="https://sap.github.io/fundamental-ngx/#/core/link">fundamental-ngx/core</a>. In Platform, Link is wrapped
```
this is fine, just fyi, you can even use `<a [routerLink]="'/core/link'">` instead

DeepakSap14 replied:
Done.

DeepakSap14 commented:
> The PR looks good to me. It doesn't cover all the points listed in #2542

Now added examples to cover Issue #2542.
extended from base class.
renamed "type" property to "linktype", as it was overlapping with native "type" for anchor tag.

KevinOkamoto commented on libs/platform/src/lib/components/link/link.component.ts:
```diff
@@ -10,51 +11,59 @@ import {
+    /** linktype of link, options standard or Emphasized, Default=standard */
     @Input()
-    type: LinkType = 'standard';
+    linktype?: LinkType = 'standard';
```
Use camel-case, change `linktype` to `linkType`.

DeepakSap14 replied:
Done

stefanoScalzo commented on libs/platform/src/lib/components/link/link.component.ts:
```diff
@@ -10,51 +11,59 @@ import {
+     * sets type to Native anchor.
+     */
     @Input()
-    get disabled(): boolean {
```
why are you removing these methods?

DeepakSap14 replied:
Link Component is extending BaseComponent. BaseComponent has disabled property defined. So removed duplication from Link Component.

fkolar replied:
Disabled state is almost on every single component

DeepakSap14 commented:
> Just camel-case `linkType`, otherwise it looks good.

changes are done and pushed.
